src/NServiceBus.Core/StartableEndpoint.cs

```diff
@@ -31,6 +31,7 @@ public StartableEndpoint(SettingsHolder settings, IBuilder builder, FeatureActiv
             DetectThrottlingConfig();
 
             var pipelineCache = new PipelineCache(builder, settings);
+            await settings.Get<TransportInfrastructure>().Start().ConfigureAwait(false);
```

Comment:
Getting it from the settings seems hacky, how about we pass it in the ctor instead?